src/cider/nrepl/middleware/util/storage.clj

```diff
@@ -0,0 +1,16 @@
+(ns cider.nrepl.middleware.util.storage)
+
+;; TODO: How should we expire the cache?
+;; Currently effectively single use storage for use with non-sync exceptions,
+;; but since this can likely be used for any type of data,
+;; perhaps we should have some formal autopurge functionality so that this
+;; doesn't grow unbounded?
+(def ^:dynamic *storage* (atom {}))
```

Comment:
Does this need to be dynamic? Most of the comment should actually the var's docstring IMO.

As for expiring the cache the last 10 exceptions or something like this and simply drop the oldest when we ran out of space.